projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -484,17 +484,21 @@ public function add_gutenberg_menus() {
 
 	/**
 	 * Add the calypso /woocommerce-installation/ menu item.
+	 *
+	 * @param array $current_plan The site's plan if they have one. This is passed from WPcom_Admin_Menu to prevent
+	 * redundant database queries.
 	 */
-	public function add_woocommerce_installation_menu() {
+	public function add_woocommerce_installation_menu( $current_plan = null ) {
```

Comment:
It feels odd to me that we have this method in the general class even though it's not used generally. Ideally the Atomic and Wpcom classes would define it themselves in the way they need it, Atomic without the current plan, WPcom with it. But since it's pre-existing, probably not necessary to hold up progress for now.

Reply:
Initially `add_woocommerce_installation_menu` only existed in `WPcom_Admin_Menu` and wpcomsh added the link itself by using `add_menu_page`. I suggested [moving the hook up](https://github.com/Automattic/jetpack/pull/21717) to `Admin_Menu` so wpcomsh could make use of it. That way we didn't duplicate the icon and link definition across two repositories. Also wpcomsh doesn't have `add_admin_menu_separator` to make use of. It might have made more sense to just re-define `add_woocommerce_installation_menu` in `Atomic_Admin_Menu`. However, at the time they would have been identical methods.